spec/lib/mahonia/identifier_builder_spec.rb

```diff
@@ -0,0 +1,42 @@
+require 'rails_helper'
+
+RSpec.describe Mahonia::IdentifierBuilder do
+  subject(:builder) { described_class.new }
+
+  it_behaves_like 'an IdentifierBuilder'
+
+  describe '#prefix' do
+    it 'has a default prefix' do
+      expect(builder.prefix).not_to be_empty
+    end
+
+    it 'accepts a prefix' do
+      prefix = 'my_pfx'
+      builder = described_class.new(prefix: prefix)
+      expect(builder.prefix).to eq prefix
+    end
+  end
+
+  describe '#build' do
+    it 'uses the prefix' do
+      expect(builder.build(hint: 'blah')).to start_with "#{builder.prefix}/"
+    end
+
+    context 'with a custom prefix' do
+      subject(:builder) { described_class.new(prefix: prefix) }
+      let(:prefix)      { 'fake_prefix' }
+
+      it 'uses the prefix' do
+        expect(builder.build(hint: 'blah')).to start_with "#{prefix}/"
+      end
+    end
+
+    it 'raises an error with no hint' do
+      expect { builder.build }.to raise_error ArgumentError
+    end
+
+    it 'uses the hint exactly, cast to uppercase' do
```

Comment:
This doesn't look cast to uppercase to me. What am I missing?

Reply:
this text is just cruft from the Epigaea port. I'll remove it.